spec/models/manageiq/providers/google/network_manager/refresh_parser_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'digest'
+
+describe ManageIQ::Providers::Google::NetworkManager::RefreshParser do
+  let(:ems) { FactoryGirl.create(:ems_google, :project => "manageiq-dev") }
+  let(:az) { FactoryGirl.create(:availability_zone_google) }
+  let(:vm) do
+    FactoryGirl.create(:vm_google,
+                       :ext_management_system => ems,
+                       :ems_ref               => 123,
+                       :availability_zone     => az)
+  end
+  # These tests use some fairly aggressive stubbing. Specifically, all of the
+  # fog calls are stubbed out. This allows us to test the refresh logic without
+  # resorting to VCR casettes.
```

Comment:
Yeah, I agree this is pretty intensive. I tried to make the specing code as readable as possible; honestly I think the goryness here is really just highlighting that the refresher class does too much and should probably be extracted out into other components.